contracts/ExpectedRate.sol

```diff
@@ -54,10 +54,27 @@ contract ExpectedRate is Withdrawable, ExpectedRateInterface, Utils2 {
 
         if (srcQty == 0) srcQty = 1;
 
-        bool didRevert = false;
+        //not needed
+        uint bestReserve;
 
-        (didRevert, expectedRate, slippageRate) = safeFindBestRate(src, dest, srcQty, usePermissionless);
-        if (didRevert) return (0, 0);
+        if (usePermissionless) {
+            (bestReserve, expectedRate) = kyberNetwork.findBestRate(src, dest, srcQty);
```

Comment:
best if can write the code as it was before safeFindBestRate was added.
is it?

Reply:
https://github.com/KyberNetwork/smart-contracts/blob/37e9bcd67dd8799ea2f571c26ecc33a6945b95aa/contracts/ExpectedRate.sol

It's the same as this, right? Minus the check for knc arbitrage.

Reply:
Great
thanks